src/OpenTelemetry/CHANGELOG.md

```diff
@@ -8,9 +8,29 @@ Notes](../../RELEASENOTES.md).
 
 * Promoted overflow attribute from experimental to stable and removed the
   `OTEL_DOTNET_EXPERIMENTAL_METRICS_EMIT_OVERFLOW_ATTRIBUTE` environment variable.
-  The overflow attribute feature is now enabled by default.
-  No internal logs will be emitted when the cardinality limit is reached, as the
-  `otel.metric.overflow` attribute can be used to tell that the limit is reached.
+
+  **Previous Behavior:**
+  By default, when the cardinality limit was reached, measurements were dropped,
+  and an internal log was emitted the first time this occurred. Users could
+  opt-in to experimental overflow attribute feature with
+  `OTEL_DOTNET_EXPERIMENTAL_METRICS_EMIT_OVERFLOW_ATTRIBUTE=true`.
+  With this setting, the SDK would use an overflow attribute
+  (`otel.metric.overflow = true`) to aggregate measurements instead of dropping
+  measurements. No internal log was emitted in this case.
+
+  **New Behavior:**
+  The SDK now always uses the overflow attribute (`otel.metric.overflow = true`)
+  to aggregate measurements when the cardinality limit is reached. The previous
+  approach of dropping measurements has been removed. No internal logs are
+  emitted when the limit is hit.
+
+  The default cardinality limit remains 2000 per metric. To adjust this globally,
+  use the `SetMaxMetricPointsPerMetricStream()` API. However, we recommend
```

Comment:
Since `SetMaxMetricPointsPerMetricStream` is marked obsolete, I think we should not recommend it here.

Suggestion:
The default cardinality limit remains 2000 per metric. To override the limit for an individual metric, use the [changing cardinality limit](../../docs/metrics/customizing-the-sdk/README.md#changing-the-cardinality-limit-for-a-metric).

Also, we can offer a new method `SetCardinalityLimit` on reader level if we want to let user change the default for entire metrics? (this is a good candidate to follow up, no need to block this PR)

Reply:
I thought you meant that API in here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5909#discussion_r1805833875.

[Changing the cardinality limit for a Metric](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/metrics/customizing-the-sdk/README.md#changing-the-cardinality-limit-for-a-metric) is still experimental though. Should we start suggesting it now?

Reply:
oh i didn't realize its still experimental!! But you plan to make it stable along with this PR (as a follow up) right?

i.e https://github.com/open-telemetry/opentelemetry-dotnet/issues/5444 should be resolved as well before the next release?

Reply:
Yes. In the last comment of that issue I also stated the same.